src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java

```diff
@@ -0,0 +1,365 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.custom.fuelsdk;
+
+import com.exacttarget.fuelsdk.ETClient;
+import com.exacttarget.fuelsdk.ETDataExtensionColumn;
+import com.exacttarget.fuelsdk.ETDataExtensionRow;
+import com.exacttarget.fuelsdk.ETExpression;
+import com.exacttarget.fuelsdk.ETFilter;
+import com.exacttarget.fuelsdk.ETResponse;
+import com.exacttarget.fuelsdk.ETResult;
+import com.exacttarget.fuelsdk.ETSdkException;
+import com.exacttarget.fuelsdk.ETSoapConnection;
+import com.exacttarget.fuelsdk.ETSoapObject;
+import com.exacttarget.fuelsdk.annotations.SoapObject;
+import com.exacttarget.fuelsdk.internal.APIObject;
+import com.exacttarget.fuelsdk.internal.RetrieveRequest;
+import com.exacttarget.fuelsdk.internal.RetrieveRequestMsg;
+import com.exacttarget.fuelsdk.internal.RetrieveResponseMsg;
+import com.exacttarget.fuelsdk.internal.Soap;
+import org.apache.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.exacttarget.fuelsdk.ETDataExtension.retrieveColumns;
+
+/**
+ * Extending class to fix the bug with Fuel-SDK on pagination.
+ */
+public class PaginationETSoapObject extends ETSoapObject {
+
+  static final int DEFAULT_PAGE_SIZE = 2500;
+  private static Logger logger = Logger.getLogger(PaginationETSoapObject.class);
+
+  /**
+   * @param client
+   * @param dataExtension
+   * @param filter
+   * @return
+   * @throws ETSdkException
+   */
+  public static ETResponse<ETDataExtensionRow> select(ETClient client,
+                                                      String dataExtension,
+                                                      ETFilter filter)
+    throws ETSdkException {
+    String name = null;
+
+    //
+    // The data extension can be specified using key or name:
+    //
+
+    ETExpression e = ETExpression.parse(dataExtension);
+    if (e.getProperty().toLowerCase().equals("key")
+      && e.getOperator() == ETExpression.Operator.EQUALS) {
+      name = e.getValue();
+      // if no columns are explicitly requested
+      // retrieve all columns
+      if (filter.getProperties().isEmpty()) {
+        filter.setProperties(retrieveColumnNames(client, name));
+      }
+    } else if (e.getProperty().toLowerCase().equals("name")
+      && e.getOperator() == ETExpression.Operator.EQUALS) {
+      name = e.getValue();
+      // if no columns are explicitly requested
+      // throw an exception
+      // because we need the key
+      // to retrieve columns
+      if (filter.getProperties().isEmpty()) {
+        throw new ETSdkException("columns must be specified "
+                                   + "when retrieving data extensions by name");
+      }
+
+    } else {
+      throw new ETSdkException("invalid data extension filter string");
+    }
+
+    ETResponse<ETDataExtensionRow> response = customRetrieve(client, "DataExtensionObject[" + name + "]",
```

Comment:
any validation on 'name'? Like, what is its value itself is null? Can that happen?

Reply:
All this code was a copy from the original ETSoapObject and extended to fix the bug.